flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/api/TableEnvironmentTest.java

```diff
@@ -110,4 +118,96 @@ public void testTableFromDescriptor() {
 
         assertEquals("fake", lookupResult.get().getTable().getOptions().get("connector"));
     }
+
+    @Test
+    public void testCreateManagedTableFromDescriptor() {
+        innerTestCreateManagedTableFromDescriptor(false, false);
+        innerTestCreateManagedTableFromDescriptor(true, false);
+        innerTestCreateManagedTableFromDescriptor(true, true);
+        innerTestCreateManagedTableFromDescriptor(false, true);
+    }
+
+    private void innerTestCreateManagedTableFromDescriptor(
+            boolean ignoreIfExists, boolean isTemporary) {
+        final TableEnvironmentMock tEnv = TableEnvironmentMock.getStreamingInstance();
+        final String catalog = tEnv.getCurrentCatalog();
+        final String database = tEnv.getCurrentDatabase();
+
+        final Schema schema = Schema.newBuilder().column("f0", DataTypes.INT()).build();
+        final String tableName = UUID.randomUUID().toString();
+        ObjectIdentifier identifier = ObjectIdentifier.of(catalog, database, tableName);
+
+        // create table
+
+        CreateTableOperation createOperation =
+                new CreateTableOperation(
+                        identifier,
+                        TableDescriptor.forManaged()
+                                .schema(schema)
+                                .option("a", "Test")
+                                .build()
+                                .toCatalogTable(),
+                        ignoreIfExists,
+                        isTemporary);
+
+        tEnv.executeInternal(createOperation);
+
+        // test ignore: create again
+        if (ignoreIfExists) {
+            tEnv.executeInternal(createOperation);
+        } else {
+            try {
```

Comment:
This entire try-catch block could be written in a nicer way using AssertJ's `assertThatThrownBy`.